presto-main/src/main/java/io/prestosql/operator/TableWriterOperator.java

```diff
@@ -227,7 +236,14 @@ public void addInput(Page page)
 
         Block[] blocks = new Block[columnChannels.size()];
         for (int outputChannel = 0; outputChannel < columnChannels.size(); outputChannel++) {
-            blocks[outputChannel] = page.getBlock(columnChannels.get(outputChannel));
+            Block block = page.getBlock(columnChannels.get(outputChannel));
+            if (block.mayHaveNull()) {
+                String columnName = notNullChannelColumnNames.get(outputChannel);
```

Comment:
Doing a map lookup for every channel could have a non-trivial cost, especially since we have to box the channel integer key. I think it would be better to use a `List<String>` with null values for the nullable columns, making the lookup basically free.

Reply:
Done in the latest force push.  However, nulls are illegal in ImmutableLists, so in TestTableWriterOperation I had to create an ArrayList.

I thought of creating Optionals, but the main cost of Map lookup is cacheline bounces, and adding Optionals will also result in cacheline bounces.